velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -114,13 +114,38 @@ VeloxToSubstraitExprConvertor::toSubstraitExpr(
   ::substrait::Expression_FieldReference* substraitFieldExpr =
       google::protobuf::Arena::CreateMessage<
           ::substrait::Expression_FieldReference>(&arena);
-
-  std::string exprName = fieldExpr->name();
-
   ::substrait::Expression_ReferenceSegment_StructField* directStruct =
       substraitFieldExpr->mutable_direct_reference()->mutable_struct_field();
+  std::string exprName = fieldExpr->name();
+  // FieldAccessTypedExpr represents one of two things: a leaf in an expression
+  // or a dereference expression(fieldExpr->isInputColumn() == false)
+  // for a leaf in an expression, find idx from child by exprName.
+  // for a dereference expression, find idx from every child by exprName.
+  if (fieldExpr->isInputColumn()) {
+    auto idx = inputType->getChildIdxIfExists(exprName);
+    if (idx.has_value()) {
+      directStruct->set_field(idx.value());
+    }
+  } else {
+    for (auto child : inputType->children()) {
+      auto rowChild = asRowType(child);
+      if (!rowChild) {
+        continue;
+      }
+      auto idxInChild = rowChild->getChildIdxIfExists(exprName);
+      if (idxInChild.has_value()) {
```

Comment:
What happens if no child has exprName field? Or if 2 or more children have exprName field?

Reply:
I think we should follow the design of function “getChildIdxIfExists”.  Find the first match from the beginning. if no child has exprName should throw an exception.

Reply:
This may lead to some flakiness. I suggest to throw if there are multiple children with exprName subfield.

Reply:
I agree with you.